### PR TITLE
SWC-6800 - Add `useNotifyNewACLUsers` hook

### PR DESCRIPTION
**Pull request by nickgros**
SWC-6800

- Add entity permission levels
- Add useNotifyNewACLUsers - Adds a hook that will be used by the entity ACL editor reimplementation to notify new users added to an ACL.
- Also consolidated public user & group IDs into a single constant.

### CODE REVIEW

File: packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+
+const sendMessageSpy = jest.spyOn(SynapseClient, 'sendMessage')
+
+describe('useNotifyNewACLUsers', () => {
+  beforeEach(() => jest.clearAllMocks())
+  beforeAll(() => server.listen())
+  afterEach(() => server.restoreHandlers())
+  afterAll(() => server.close())
+
+  it('Sends a notification to new users', async () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts:44]: Two basic integration tests to verify 1) the hook sends a notification when it should and 2) it does not when it shouldn't.

Unit tests capture complexity below

File: packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+    })
+
+    await waitFor(() => {
+      expect(result.current.isPending).toBe(false)
+      expect(sendMessageSpy).toHaveBeenCalledTimes(0)
+    })
+  })
+})
+
+describe('shouldNotifyUserInNewResourceAccess', () => {
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts:131]: Unit test the utility function to validate complex logic

File: packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx [hunk 3]
```diff
@@ -10,35 +10,27 @@ import {
...
-import { FeatureFlagEnum } from '@sage-bionetworks/synapse-types'
 
 function isPublic(bundle: EntityBundle): boolean {
   return bundle.benefactorAcl.resourceAccess.some(ra => {
-    return (
-      ra.principalId === AUTHENTICATED_PRINCIPAL_ID ||
-      ra.principalId === PUBLIC_PRINCIPAL_ID ||
-      ra.principalId === ANONYMOUS_PRINCIPAL_ID
-    )
+    return PUBLIC_PRINCIPAL_IDS.includes(ra.principalId)
```
Comment (nickgros) [inline packages/synapse-react-client/src/components/EntityBadgeIcons/EntityBadgeIcons.tsx:33]: use new constant

File: packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts (hunk 1 above)
Comment (jay-hodgson) [reply packages/synapse-react-client/src/components/EntityAclEditor/useNotifyNewACLUsers.test.ts:44]: Excellent set of tests

**Review by jay-hodgson — APPROVED** [review]

**Pull request merged**